File: 792-binary-search/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int l =0 , r = nums.length;

        while(l<r){
            int mid = l+(r-l)/2;
            if(nums[mid]>=target){
                r = mid;
            }
            else{
                l = mid+1;
            }
        }

        if (l<nums.length && nums[l] == target){
            return l;
        }else{
            return -1;
        }
    }
}

File: 792-binary-search/binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l,r = 0, len(nums)-1

        while l<r:
            m = (l+r)>>1
            if nums[m] >= target:
                r = m
            else:
                l = m+1

        return l if nums[l] == target else -1
            